strawberryfields/tdm/tdmprogram.py

```diff
@@ -580,7 +580,7 @@ def unroll(self, shots):
         q = self.register
 
         sm = []
-        for i in range(len(self.N)):
+        for i, _ in enumerate(self.N):
```

Comment:
I would not have touched this line vy default, but codeFactor (one of the CI tests) was failing because of the old version of this line. Happy to revert to the original if you help me get past the failed codeFactor check.

Reply:
I'm actually not sure which I prefer. I would have used the former (which I believe might be slightly faster, but not by much), although I actually now thinking that using `enumerate` looks cleaner. I'm happy to keep your change here.